cron.go

```diff
@@ -352,3 +428,30 @@ func (s *Schedule) union(s1 *Schedule) {
 		s.b[i] |= s1.b[i]
 	}
 }
+
+type hashedFields struct {
+	rng    *rand.Rand
+	fields [5]int
+}
+
+// newHashedFields generates random schedule values for the given seed.
+// The same schedule values are generated given the same seed.
+func newHashedFields(seed uint64) *hashedFields {
```

Comment:
ISTM that it would simplify things to generate all the random numbers as needed rather than pre-generate five of them.